### PR TITLE
feat: implement multiple catalog AST

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: 蔡略 <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR implements the AST parser and data type part of multiple catalogs implementation

A split part of #8503
Implementation part of #8255

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]

File: src/query/ast/tests/it/testdata/statement.txt [hunk 1]
```diff
@@ -743,6 +743,24 @@ UseDatabase {
 }
 
 
+---------- Input ----------
+create catalog ctl type=hive connection=(url='<hive-meta-store>' thrift_protocol='binary');
+---------- Output ---------
+CREATE CATALOG ctl TYPE='HIVE'
```
Comment (andylokandy) [inline src/query/ast/tests/it/testdata/statement.txt:749]: display is incomplete

File: src/query/ast/src/parser/statement.rs [hunk 2]
```diff
@@ -178,6 +179,48 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
         },
     );
 
+    // catalogs
+    let show_catalogs = map(
+        rule! {
+            SHOW ~ ( CATALOGS ) ~ #show_limit?
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:185]:
```suggestion
            SHOW ~ CATALOGS ~ #show_limit?
```

File: src/query/ast/src/parser/statement.rs [hunk 3]
```diff
@@ -178,6 +179,48 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
...
+        },
+        |(_, _, limit)| Statement::ShowCatalogs(ShowCatalogsStmt { limit }),
+    );
+    let show_create_catalog = map(
+        rule! {
+            SHOW ~ CREATE ~ ( CATALOG ) ~ #ident
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:191]:
```suggestion
            SHOW ~ CREATE ~ CATALOG ~ #ident
```

File: src/query/ast/src/parser/statement.rs [hunk 4]
```diff
@@ -178,6 +179,48 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
...
+        },
+        |(_, _, _, catalog)| Statement::ShowCreateCatalog(ShowCreateCatalogStmt { catalog }),
+    );
+    // TODO: support `COMMENT` in create catalog
+    let create_catalog = map(
+        rule! {
+            CREATE ~ ( CATALOG ) ~ ( IF ~ NOT ~ EXISTS )?
+            ~ #ident
+            ~ ( TYPE ~ "=" ~ #catalog_type )
+            ~ ( CONNECTION ~ "=" ~ #options )
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:201]:
```suggestion
            CREATE ~ CATALOG ~ ( IF ~ NOT ~ EXISTS )?
            ~ #ident
            ~ TYPE ~ "=" ~ #catalog_type
            ~ CONNECTION ~ "=" ~ #options
```

File: src/query/ast/src/parser/statement.rs [hunk 5]
```diff
@@ -178,6 +179,48 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
...
+                if_not_exists: opt_if_not_exists.is_some(),
+                catalog_name: catalog.to_string(),
+                catalog_type: ty,
+                options,
+            })
+        },
+    );
+    let drop_catalog = map(
+        rule! {
+            DROP ~ ( CATALOG ) ~ ( IF ~ EXISTS )? ~ #ident
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:214]:
```suggestion
            DROP ~ CATALOG ~ ( IF ~ EXISTS )? ~ #ident
```

**Review by andylokandy — APPROVED** [review]

**Pull request merged**